addon-test-support/-private/ember-exam-qunit-test-loader.js

```diff
@@ -0,0 +1,154 @@
+    let partitions = this._urlParams.get('partition');
+    let split = parseInt(this._urlParams.get('split'), 10);
+
+    split = isNaN(split) ? 1 : split;
+
+    if (partitions === undefined) {
+      partitions = [1];
+    } else if (!Array.isArray(partitions)) {
+      partitions = [partitions];
+    }
+
+    super.loadModules();
+
+    if (loadBalance && this._testem) {
+      this.setupLoadBalanceHandlers();
+      this._testModules = splitTestModules(
+        weightTestModules(this._testModules),
+        split,
+        partitions
+      );
+      this._testem.emit(
+        'testem:set-modules-queue',
+        this._testModules,
+        browserId
+      );
+    } else {
+      this._testModules = splitTestModules(
+        this._testModules,
+        split,
+        partitions
+      );
+      this._testModules.forEach((moduleName) => {
+        super.require(moduleName);
+        super.unsee(moduleName);
+      });
+    }
+  }
+
+  /**
+   * Allow loading one module at a time.
+   *
+   * @param {string} moduleName
+   */
+  loadIndividualModule(moduleName) {
+    if (moduleName === undefined) {
+      throw new Error(
+        'Failed to load a test module. `moduleName` is undefined in `loadIndividualModule`.'
+      );
+    }
+    super.require(moduleName);
+    super.unsee(moduleName);
+  }
+
+  /**
+   * setupLoadBalanceHandlers() registers QUnit callbacks needed for the load-balance option.
+   */
+  setupLoadBalanceHandlers() {
+    // nextModuleAsyncIterator handles the async testem events
+    // it returns an element of {value: <moduleName>, done: boolean}
+    const nextModuleAsyncIterator = new AsyncIterator(this._testem, {
+      request: 'testem:next-module-request',
+      response: 'testem:next-module-response'
+    });
+
+    const nextModuleHandler = () => {
+      return new Promise((resolve, reject) => {
+        nextModuleAsyncIterator
+          .next()
+          .then((response) => {
+            if (!response.done) {
+              const moduleName = response.value;
+              this.loadIndividualModule(moduleName);
+
+              // if no tests were added, request the next module
+              if (this._qunit.config.queue.length === 0) {
+                resolve(nextModuleHandler());
+              }
+            }
+            resolve();
+          })
+          .catch(err => {
+            reject(`Failed to get a next test module to load. - ${err}`);
+          });
+      });
+    };
+
+    // it registers qunit begin callback to ask for a next test moudle to execute when the test suite begins.
+    // By default ember-qunit adds `Ember.onerror` test to a qunit processing queue and once the test is complete it execute _qunit.moduleDone callback.
+    // However, when `setupEmberOnerrorValidation: false` is passed the test is disabled and _qunit.begin callback needs to request a next test module to run.
+    this._qunit.begin(() => {
+      return nextModuleHandler.call(this);
```

Comment:
nit: `call(this)` is misleading as the rebound `this` will be discarded as `nextModuleHandler` is an `=>` function, which already preserves `this`


Example:
```js
a = () => console.log(this)
a(); 
> Window {postMessage: ƒ, blur: ƒ, focus: ƒ, close: ƒ, parent: Window, …} // global this

a.call('i am new this')
> Window {postMessage: ƒ, blur: ƒ, focus: ƒ, close: ƒ, parent: Window, …} // same this as before
```